### ISSUE TITLE
Upgrade to 11.2-RELEASE: Run '/tmp/tmp70lcmdme fetch' first.

### ISSUE THREAD

**Issue by serge-vk** [iocage/iocage#579]
- [x] Supply `iocage --version`
Version	0.9.10 2017/12/22
- [x] Supply the commands used, along with any steps to recreate it.
```
iocage fetch # select 11.2-RELEASE
```
- [x] Provide the output from the command you issued.
<details>
<summary> ...Run '/tmp/tmp70lcmdme fetch' first. </summary>

```
Fetching: 11.2-RELEASE

Extracting: base.txz...
Extracting: lib32.txz...
Extracting: doc.txz...
Extracting: src.txz...

* Updating 11.2-RELEASE to the latest patch level...
Looking up update.FreeBSD.org mirrors... 3 mirrors found.
Fetching metadata signature for 11.2-RELEASE from update4.freebsd.org... done.
Fetching metadata index... done.
Inspecting system... done.
Preparing to download files... done.

No updates needed to update system to 11.2-RELEASE-p0.
No updates are available to install.
Run '/tmp/tmp70lcmdme fetch' first.
```
- [x] Supply what you expected the result or output to be
Fully fetched release
</details>

- [x] Checked that the problem has not already been fixed on `master` if using
 a stable release.

I tried locating these temporary files, but there are no such names on my system after the release is fetched. When I upgrade a jail with `iocage upgrade jailname -r 11.2-RELEASE` and start the upgraded jail, `freebsd-version` still reports that it is at 11.1-RELEASE-p9. I tried `iocage update jailname` as well. The output is similar:

<details>
<summary> ...Run '/tmp/tmptonyeuac fetch' first. </summary>
 
```
* Updating 11.2-RELEASE to the latest patch level...
Looking up update.FreeBSD.org mirrors... 3 mirrors found.
Fetching metadata signature for 11.2-RELEASE from update6.freebsd.org... done.
Fetching metadata index... done.
Inspecting system... done.
Preparing to download files... done.

No updates needed to update system to 11.2-RELEASE-p0.
No updates are available to install.
Run '/tmp/tmptonyeuac fetch' first.

jailname has been updated successfully.
```
</details>

The problem is that all the userland seems to be still from 11.1. For example, when I run `clang -v`, it reports 4.0.0, whereas on the host it is, as expected 6.0.0. On the host system, the upgrade went without problems and I never saw a message about running a temporary file to fetch something.

I wonder how to get around this temp file problem, thanks.

**Comment by skarekrow** [comment]
The temp file is expected. Please try the master branch on GitHub, upgrade had a lot of fixes.

**Comment by serge-vk** [comment]
Hi there, thank you for your response. I only had time to follow your advice now. Unfortunately, I could not build iocage from the repo because libzfs fails with these errors:
```
libzfs.c:44888:73: error: use of undeclared identifier 'POOL_SCRUB_NORMAL'
        __pyx_v_ret = zpool_scan(__pyx_v_self->handle, POOL_SCAN_SCRUB, POOL_SCRUB_NORMAL);
                                                                        ^
libzfs.c:45083:72: error: use of undeclared identifier 'POOL_SCRUB_NORMAL'
        __pyx_v_ret = zpool_scan(__pyx_v_self->handle, POOL_SCAN_NONE, POOL_SCRUB_NORMAL);
                                                                       ^
libzfs.c:51060:86: error: too many arguments to function call, expected 3, have 4
        __pyx_v_ret = lzc_send_space(__pyx_v_c_name, __pyx_v_cfromname, __pyx_int_0, (&__pyx_v_space));
                      ~~~~~~~~~~~~~~                                                 ^~~~~~~~~~~~~~~~

```
I wonder what I am doing incorrectly. Forgot to say that I tried building in a jail running 11.1-RELEASE-p9. Also, this seems to be related to this: https://github.com/freenas/py-libzfs/issues/1

**Comment by serge-vk** [comment]
I tried compiling under 11.2 and it worked. But I think, it had better be able to build from 11.1, because otherwise this could create a catch-22 situation (it did for me, because I use a jail to build everything and now I cannot build iocage to try updating the jail).

Now, I have just run `iocage fetch` from master branch and got precisely the same output as I first reported. There is also no temp file named in the output anywhere on my system. 

There is, unfortunately, no difference from the Ports version of iocage.
Edit: in case you wondered, I tried also `iocage update jailname` with the same result as before. `freebsd-version` still reports the previous release.

**Comment by atl** [comment]
Just wanted to +1 this issue, as I'm seeing it on `Version 1.0 ALPHA`.

**Comment by rduffner** [comment]
I also see this on 1.0 on FreeBSD 12.
``
No updates are available to install.

Run '/tmp/tmp72bxvo68 fetch' first.
``


Trying to upgrade a 11.2 jail to 12.0 (host is 12, upgraded from 11.2)

**Comment by skarekrow** [comment]
The core issue was upgrade here, the message is expected. This was solved in a recent PR #817
